### PR TITLE
Eslint 6 support (major)

### PR DESCRIPTION
**Pull request by DullReferenceException**
- Enable `eslint@6`
- Fix peerDependencies to align with devDependencies
- Remove unused dependency

### CODE REVIEW

File: packages/eslint-config-godaddy/package.json [hunk 1]
```diff
@@ -24,16 +24,15 @@
     "eslint-godaddy": "bin/eslint-godaddy"
   },
   "peerDependencies": {
-    "eslint": "^5.8.0",
-    "eslint-plugin-json": "^1.2.0",
-    "eslint-plugin-mocha": "^5.2.0"
+    "eslint": "^6.1.0",
```
Comment (msluther) [inline packages/eslint-config-godaddy/package.json]: Should this change across all the packages at once?

Comment (DullReferenceException) [reply packages/eslint-config-godaddy/package.json]: I'm not sure if I can do that, sadly. The problem is that some of these packages have dependencies on other packages in this repo, and those packages have `eslint` peer dependencies which would be disallowed. So I'm going to have to publish this one first, then update those dependent on it afterward (in fact, I've got one queued up that does just that).

Comment (scommisso) [reply packages/eslint-config-godaddy/package.json]: Can do a major version bump of this library, then consumers who want ESlint5 can remain on the previous version, right?

Comment (DullReferenceException) [reply packages/eslint-config-godaddy/package.json]: @scommisso: that's the plan, to make this a major bump (going to do that post-merge).

**Pull request merged**